datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -649,7 +649,9 @@ impl OptimizerRule for PushDownFilter {
                 let mut push_predicates = vec![];
                 for expr in predicates {
                     let cols = expr.to_columns()?;
-                    if cols.iter().all(|c| group_expr_columns.contains(c)) {
+                    if !expr.contains_hidden_columns()
+                        && cols.iter().all(|c| group_expr_columns.contains(c))
+                    {
```

Comment:
`hiddent columns` may influent optimizer rule, it may cause some bug.
We should add more test to cover it in the following PR.